soh/src/code/z_player_lib.c

```diff
@@ -1188,6 +1207,27 @@ void func_8008F87C(PlayState* play, Player* this, SkelAnime* skelAnime, Vec3f* p
 s32 Player_OverrideLimbDrawGameplayCommon(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
     Player* this = (Player*)thisx;
 
+    if (CVarGetInteger("gChildHoldsHylianShield", 0) && CVarGetInteger("gRotateScaleChildHylianShield", 0) &&
+        LINK_IS_CHILD) {
+        if (limbIndex == PLAYER_LIMB_SHEATH) {
+            if ((this->currentShield == PLAYER_SHIELD_HYLIAN &&
+                 (gSaveContext.equips.buttonItems[0] == ITEM_SWORD_KOKIRI ||
+                  gSaveContext.equips.buttonItems[0] == ITEM_NONE ||
+                  gSaveContext.equips.buttonItems[0] == ITEM_STICK)) &&
+                ((this->sheathType == PLAYER_MODELTYPE_SHEATH_18) ||
+                 (this->sheathType == PLAYER_MODELTYPE_SHEATH_19))) {
+                Matrix_Translate(218, 0, 62, MTXMODE_APPLY);
+                Matrix_Scale(0.8, 0.8, 0.8, MTXMODE_APPLY);
+            }
+        }
+        if (limbIndex == PLAYER_LIMB_R_HAND) {
+            if (this->currentShield == PLAYER_SHIELD_HYLIAN && sRightHandType == PLAYER_MODELTYPE_RH_SHIELD &&
+                gSaveContext.equips.buttonItems[0] == ITEM_SWORD_KOKIRI) {
+                Matrix_Scale(0.8, 0.8, 0.8, MTXMODE_APPLY);
+            }
+        }
+    }
```

Comment:
Maybe not for this PR but maybe we could add a hook here for player override limb draw common so that future modifications like this dont stack up in this function